### DISCUSSION TITLE
Non standard modulation schemes testing implementation

### DISCUSSION THREAD

**Discussion by jamie0102** [srsran/srsRAN_Project#591] [General Help]
Hello,

I'm considering implementing some non-standard modulation schemes, specifically SEFDM, in the gNB and UE for a university project. From a coding standpoint, I'm a bit unsure of where to start. Any help, advice, or pointers in the right direction would be greatly appreciated. I'm also open to ideas on how this could be achieved, if possible at all.

Many thanks,

Jamie

**Comment by pgawlowicz** [comment]
Hi,
If I understand correctly, you need to replace FFT with FrFT on the TX side.

You probably can start with PDSCH:
- srsgnb: https://github.com/srsran/srsRAN_Project/blob/main/lib/phy/lower/modulation/ofdm_modulator_impl.cpp#L90
- srsUE: https://github.com/srsran/srsRAN_4G/blob/master/srsue/src/phy/nr/cc_worker.cc#L543
